File: Python/Table.py
# Saya Alif Faturahman Firdaus (2107377) mengerjakan Latihan Praktikum 2 dalam mata 
# kuliah DPBO untuk keberkahan-Nya maka saya tidak melakukan kecurangan seperti yang 
# telah dispesifikasikan. Aamiin.

# ----- Latihan Python ----- #

# Membuat kelas table
class Table:
    # Membuat atribut Private
    # banyaknya baris dan kolom
    __baris = 0
    __kolom = 0

    # Constructor with default initialization
    def __init__(self, baris=0, kolom=0):
        self.__baris = baris
        self.__kolom = kolom

    # Setter and Getter methods for private attributes
    # Set Baris
    def setBaris(self, baris):
        self.__baris = baris

    # Get Baris
    def getBaris(self):
        return self.__baris

    # Set KOlom
    def setKolom(self, kolom):
        self.__kolom = kolom

    # Get KOlom
    def getKolom(self):
        return self.__kolom

    # Membuat Metode untuk buat tabel
    def buatBaris(self, isi, batas, idx, idProduct, name, brand, price, size, material, gender, color, sleeve_type):
        i = 0
        j = 0

        # jika index = 0 (membuat kerangka tabel atas)
        if idx == 0:
            print("+", end="")
            # Membuat perulangan sebanyak 10x(banyak jumlah elemen dalam 1 baris)
            for i in range(10):
                # jika i = 0
                if i == 0:
                    # Membuat perulangan untuk garis sebanyak 4x("No" + 2spasi)
                    for j in range(4):
                        print("-", end="")
                # jika i = 1
                if i == 1:
                    # Membuat perulangan untuk garis sebanyak kata terpanjang pada baris nama + 2spasi
                    print("+", end="")
                    for j in range(idProduct + 2):
                        print("-", end="")
                # jika i = 2
                if i == 2:
                    print("+", end="")
                    # Membuat perulangan untuk garis sebanyak kata terpanjang pada baris name + 2spasi
                    for j in range(name + 2):
                        print("-", end="")
                # jika i = 3
                if i == 3:
                    print("+", end="")
                    # Membuat perulangan untuk garis sebanyak kata terpanjang pada baris brand + 2spasi
                    for j in range(brand + 2):
                        print("-", end="")
                # jika i = 4
                if i == 4:
                    print("+", end="")
                    # Membuat perulangan untuk garis sebanyak kata terpanjang pada baris price + 2spasi
                    for j in range(price + 2):
                        print("-", end="")
                # jika i = 5
                if i == 5:
                    print("+", end="")
                    # Membuat perulangan untuk garis sebanyak kata terpanjang pada baris size + 2spasi
                    for j in range(size + 2):
                        print("-", end="")
                # jika i = 6
                if i == 6:
                    print("+", end="")
                    # Membuat perulangan untuk garis sebanyak kata terpanajgn pada baris material + 2spasi
                    for j in range(material + 2):
                        print("-", end="")
                # jika i = 7
                if i == 7:
                    print("+", end="")
                    # Membuat perulangan  untuk garis sebanyak kata terpanajgn pada gender + 2spasi
                    for j in range(gender + 2):
                        print("-", end="")
                # jika i = 8
                if i == 8:
                    print("+", end="")
                    # Membuat perulangan  untuk garis sebanyak kata terpanajgn pada color + 2spasi
                    for j in range(color + 2):
                        print("-", end="")
                # jika i = 9
                if i == 9:
                    print("+", end="")
                    # Membuat perulangan  untuk garis sebanyak kata terpanajgn pada sleeve_type + 2spasi
                    for j in range(sleeve_type + 2):
                        print("-", end="")
            print("+")

        # buat baris untuk judul
        if idx == 0:
            print("|No  ", end="")
            print("|ID", end="")
            for j in range(idProduct):
                # meminta uoutputan berupa spasi
                print(" ", end="")
            print("|Name", end="")
            for j in range(name - 2):
                # meminta uoutputan berupa spasi
                print(" ", end="")
            print("|Brand", end="")
            for j in range(brand - 3):
                # meminta uoutputan berupa spasi
                print(" ", end="")
            print("|Price", end="")
            for j in range(price - 3):
                # meminta uoutputan berupa spasi
                print(" ", end="")
            print("|Size", end="")
            for j in range(size - 2):
                # meminta uoutputan berupa spasi
                print(" ", end="")
            print("|Material", end="")
            for j in range(material - 6):
                # meminta uoutputan berupa spasi
                print(" ", end="")
            print("|Gender", end="")
            for j in range(gender - 4):
                # meminta uoutputan berupa spasi
                print(" ", end="")
            print("|Color", end="")
            for j in range(color - 3):
                # meminta uoutputan berupa spasi
                print(" ", end="")
            print("|Sleeve Type", end="")
            for j in range(sleeve_type - 9):
                # meminta uoutputan berupa spasi
                print(" ", end="")
            print("|")

        # jika index = 0 (membuat kerangka tabel bawah judul)
        if idx == 0:
            #  Ujung kiri tabel
            print("+", end="")
            for i in range(10):
                if i == 0:
                    # Membuat perulangan untuk garis sebanyak 4
                    for j in range(4):
                        print("-", end="")
                if i == 1:
                    print("+", end="")
                    # Membuat perulangan untuk garis sebanyak kata terpanjang pada baris nama + 2spasi
                    for j in range(idProduct + 2):
                        print("-", end="")
                if i == 2:
                    print("+", end="")
                    # Membuat perulangan untuk garis sebanyak kata terpanjang pada baris name + 2spasi
                    for j in range(name + 2):
                        print("-", end="")
                if i == 3:
                    print("+", end="")
                    # Membuat perulangan untuk garis sebanyak kata terpanjang pada baris brand + 2spasi
                    for j in range(brand + 2):
                        print("-", end="")
                if i == 4:
                    print("+", end="")
                    # Membuat perulangan untuk garis sebanyak kata terpanjang pada baris price + 2spasi
                    for j in range(price + 2):
                        print("-", end="")
                if i == 5:
                    print("+", end="")
                    # Membuat perulangan untuk garis sebanyak kata terpanjang pada baris size + 2spasi
                    for j in range(size + 2):
                        print("-", end="")
                if i == 6:
                    print("+", end="")
                    # Membuat perulangan untuk garis sebanyak kata terpanjang pada baris material + 2spasi
                    for j in range(material + 2):
                        print("-", end="")
                if i == 7:
                    print("+", end="")
                    # Membuat perulangan untuk garis sebanyak kata terpanjang pada baris gender + 2spasi
                    for j in range(gender + 2):
                        print("-", end="")
                if i == 8:
                    print("+", end="")
                    # Membuat perulangan untuk garis sebanyak kata terpanjang pada baris color + 2spasi
                    for j in range(color + 2):
                        print("-", end="")
                if i == 9:
                    print("+", end="")
                    # Membuat perulangan untuk garis sebanyak kata terpanjang pada baris sleeve_type + 2spasi
                    for j in range(sleeve_type + 2):
                        print("-", end="")
            # penutup table(ujung kanan)
            print("+")

        # buat baris isi kolom atribut
        for i in range(10):
            if i == 0:
                # tampil nomor
                print("|" + isi[i] + "   ", end="")
            elif i == 1:
                # tampil nama
                print("|" + isi[i], end="")
                for j in range(idProduct - len(isi[i]) + 2):
                    print(" ", end="")
            elif i == 2:
                # tampil name
                print("|" + isi[i], end="")
                for j in range(name - len(isi[i]) + 2):
                    print(" ", end="")
            elif i == 3:
                # tampil brand
                print("|" + isi[i], end="")
                for j in range(brand - len(isi[i]) + 2):
                    print(" ", end="")
            elif i == 4:
                # tampil price
                print("|" + isi[i], end="")
                for j in range(price - len(isi[i]) + 2):
                    print(" ", end="")
            elif i == 5:
                # tampil size
                print("|" + isi[i], end="")
                for j in range(size - len(isi[i]) + 2):
                    print(" ", end="")
            elif i == 6:
                # tampil material
                print("|" + isi[i], end="")
                for j in range(material - len(isi[i]) + 2):
                    print(" ", end="")
            elif i == 7:
                # tampil gender
                print("|" + isi[i], end="")
                for j in range(gender - len(isi[i]) + 2):
                    print(" ", end="")
            elif i == 8:
                # tampil color
                print("|" + isi[i], end="")
                for j in range(color - len(isi[i]) + 2):
                    print(" ", end="")
            elif i == 9:
                # tampil sleeve_type
                print("|" + isi[i], end="")
                for j in range(sleeve_type - len(isi[i]) + 2):
                    print(" ", end="")
            # Membuat garis penutup table pada setiap atibut baris
            if i == (9):
                print("|")
        # print("")

        # Membuat kerangka tabel pada baris tearkhir
        if idx == batas - 1:
            print("+", end="")
            for i in range(10):
                if i == 0:
                    # Membuat perulangan sebanyak 4x("No" + 2spasi)
                    for j in range(4):
                        print("-", end="")
                if i == 1:
                    print("+", end="")
                    # Membuat perulangan sebanyak kata terpanjang pada baris idProduct + 2spasi
                    for j in range(idProduct + 2):
                        print("-", end="")
                if i == 2:
                    print("+", end="")
                    # Membuat perulangan sebanyak kata terpanjang pada baris name + 2spasi
                    for j in range(name + 2):
                        print("-", end="")
                if i == 3:
                    print("+", end="")
                    # Membuat perulangan sebanyak kata terpanjang pada baris brand + 2spasi
                    for j in range(brand + 2):
                        print("-", end="")
                if i == 4:
                    print("+", end="")
                    # Membuat perulangan sebanyak kata terpanjang pada baris price + 2spasi
                    for j in range(price + 2):
                        print("-", end="")
                if i == 5:
                    print("+", end="")
                    # Membuat perulangan sebanyak kata terpanjang pada baris size + 2spasi
                    for j in range(size + 2):
                        print("-", end="")
                if i == 6:
                    print("+", end="")
                    # Membuat perulangan sebanyak kata terpanajgn pada baris material + 2spasi
                    for j in range(material + 2):
                        print("-", end="")
                if i == 7:
                    print("+", end="")
                    # Membuat perulangan sebanyak kata terpanajgn pada gender + 2spasi
                    for j in range(gender + 2):
                        print("-", end="")
                if i == 8:
                    print("+", end="")
                    # Membuat perulangan sebanyak kata terpanajgn pada color + 2spasi
                    for j in range(color + 2):
                        print("-", end="")
                if i == 9:
                    print("+", end="")
                    # Membuat perulangan sebanyak kata terpanajgn pada sleeve_type + 2spasi
                    for j in range(sleeve_type + 2):
                        print("-", end="")
            # ujung kanan table
            print("+")

File: C++/Table.cpp
/* 
    Saya Alif Faturahman Firdaus (2107377) mengerjakan Latihan Praktikum 2 dalam mata 
    kuliah DPBO untuk keberkahan-Nya maka saya tidak melakukan kecurangan seperti yang 
    telah dispesifikasikan. Aamiin.
*/

// ----- Latihan C++ ----- //

// Import library.
#include <iostream>
#include <string>

// Using standard namespace.
using namespace std;

// Membuat kelas Table
class Table
{
private:
    // banyaknya baris
    int baris;
    // banyaknya kolom
    int kolom;

public:
    Table() {} // konstruktor Table
    Table(int baris, int kolom)
    {
        // konstruktor langsung isi attribut
        this->baris = baris;
        this->kolom = kolom;
    }

    // Setter and Getter methods for private attributes
    // Set Baris
    void setBaris(int baris)
    {
        this->baris = baris;
    }

    // Get Baris
    int getBaris()
    {
        return baris;
    }

    // Set Kolom
    void setKolom(int kolom)
    {
        this->kolom = kolom;
    }

    // Get Kolom
    int getKolom()
    {
        return kolom;
    }

    // Membuat Metode untuk buat tabel
    void buatBaris(string isi[], int batas, int idx, int idProduct, int name, int brand, int price, int size, int material, int gender, int color, int sleeve_type)
    {
        int i = 0, j = 0;

        // Jika index = 0 (membuat kerangka tabel atas)
        if (idx == 0)
        {
            cout << "+";
            // Membuat perulangan sebanyak 9x (banyak jumlah elemen dalam 1 baris)
            for (i = 0; i < 10; i++)
            {
                // Jika i = 0
                if (i == 0)
                {
                    // Membuat perulangan untuk garis sebanyak 4x("No" + 2 spasi)
                    for (j = 0; j < 4; j++)
                    {
                        cout << "-";
                    }
                }
                // Jika i = 1
                if (i == 1)
                {
                    cout << "+";
                    // Membuat perulangan untuk garis sebanyak kata terpanjang pada baris nama + 2 spasi
                    for (j = 0; j < idProduct + 2; j++)
                    {
                        cout << "-";
                    }
                }
                // Jika i = 1
                if (i == 2)
                {
                    cout << "+";
                    // Membuat perulangan untuk garis sebanyak kata terpanjang pada baris name + 2 spasi
                    for (j = 0; j < name + 2; j++)
                    {
                        cout << "-";
                    }
                }
                // Jika i = 3
                if (i == 3)
                {
                    cout << "+";
                    // Membuat perulangan untuk garis sebanyak kata terpanjang pada baris brand + 2 spasi
                    for (j = 0; j < brand + 2; j++)
                    {
                        cout << "-";
                    }
                }
                // Jika i = 4
                if (i == 4)
                {
                    cout << "+";
                    // Membuat perulangan untuk garis sebanyak kata terpanjang pada baris price + 2 spasi
                    for (j = 0; j < price + 2; j++)
                    {
                        cout << "-";
                    }
                }
                // Jika i = 5
                if (i == 5)
                {
                    cout << "+";
                    // Membuat perulangan untuk garis sebanyak kata terpanjang pada baris size + 2 spasi
                    for (j = 0; j < size + 2; j++)
                    {
                        cout << "-";
                    }
                }
                // Jika i = 6
                if (i == 6)
                {
                    cout << "+";
                    // Membuat perulangan untuk garis sebanyak kata terpanajgn pada baris material + 2 spasi
                    for (j = 0; j < material + 2; j++)
                    {
                        cout << "-";
                    }
                }
                // Jika i = 7
                if (i == 7)
                {
                    cout << "+";
                    // Membuat perulangan  untuk garis sebanyak kata terpanajgn pada gender + 2 spasi
                    for (j = 0; j < gender + 2; j++)
                    {
                        cout << "-";
                    }
                }
                // Jika i = 8
                if (i == 8)
                {
                    cout << "+";
                    // Membuat perulangan  untuk garis sebanyak kata terpanajgn pada color + 2 spasi
                    for (j = 0; j < color + 2; j++)
                    {
                        cout << "-";
                    }
                }
                // Jika i = 9
                if (i == 9)
                {
                    cout << "+";
                    // Membuat perulangan  untuk garis sebanyak kata terpanajgn pada sleeve_type + 2 spasi
                    for (j = 0; j < sleeve_type + 2; j++)
                    {
                        cout << "-";
                    }
                }
            }
            // ujung table
            cout << "+";
            cout << "" << endl;
        }

        // Membuat baris untuk judul
        if (idx == 0)
        {
            cout << "|No  ";
            cout << "|ID";
            for (j = 0; j < idProduct; j++)
            {
                // Meminta output berupa spasi
                cout << " ";
            }
            cout << "|Name";
            for (j = 0; j < name - 2; j++)
            {
                // Meminta output berupa spasi
                cout << " ";
            }
            cout << "|Brand";
            for (j = 0; j < brand - 3; j++)
            {
                // Meminta output berupa spasi
                cout << " ";
            }
            cout << "|Price";
            for (j = 0; j < price - 3; j++)
            {
                // Meminta output berupa spasi
                cout << " ";
            }
            cout << "|Size";
            for (j = 0; j < size - 2; j++)
            {
                // Meminta output berupa spasi
                cout << " ";
            }
            cout << "|Material";
            for (j = 0; j < material - 6; j++)
            {
                // Meminta output berupa spasi
                cout << " ";
            }
            cout << "|Gender";
            for (j = 0; j < gender - 4; j++)
            {
                // Meminta output berupa spasi
                cout << " ";
            }
            cout << "|Color";
            for (j = 0; j < color - 3; j++)
            {
                // Meminta output berupa spasi
                cout << " ";
            }
            cout << "|Sleeve Type";
            for (j = 0; j < sleeve_type - 9; j++)
            {
                // Meminta output berupa spasi
                cout << " ";
            }
            // Penutup pada tiap baris
            cout << "|";
            cout << "" << endl;
        }

        // Jika index = 0 (membuat kerangka tabel bawah judul)
        if (idx == 0)
        {
            // Ujung kiri table
            cout << "+";
            for (i = 0; i < 10; i++)
            {
                if (i == 0)
                {
                    // Membuat perulangan untuk garis sebanyak 4
                    for (j = 0; j < 4; j++)
                    {
                        cout << "-";
                    }
                }
                if (i == 1)
                {
                    cout << "+";
                    // Membuat perulangan untuk garis sebanyak kata terpanjang pada baris nama + 2 spasi
                    for (j = 0; j < idProduct + 2; j++)
                    {
                        cout << "-";
                    }
                }
                if (i == 2)
                {
                    cout << "+";
                    // Membuat perulangan untuk garis sebanyak kata terpanjang pada baris name + 2 spasi
                    for (j = 0; j < name + 2; j++)
                    {
                        cout << "-";
                    }
                }
                if (i == 3)
                {
                    cout << "+";
                    for (j = 0; j < brand + 2; j++)
                    {
                        // Membuat perulangan untuk garis sebanyak kata terpanjang pada baris brand + 2 spasi
                        cout << "-";
                    }
                }
                if (i == 4)
                {
                    cout << "+";
                    for (j = 0; j < price + 2; j++)
                    {
                        // Membuat perulangan untuk garis sebanyak kata terpanjang pada baris price + 2 spasi
                        cout << "-";
                    }
                }
                if (i == 5)
                {
                    cout << "+";
                    // Membuat perulangan untuk garis sebanyak kata terpanjang pada baris size + 2 spasi
                    for (j = 0; j < size + 2; j++)
                    {
                        cout << "-";
                    }
                }
                if (i == 6)
                {
                    cout << "+";
                    // Membuat perulangan untuk garis sebanyak kata terpanjang pada baris material + 2 spasi
                    for (j = 0; j < material + 2; j++)
                    {
                        cout << "-";
                    }
                }
                if (i == 7)
                {
                    cout << "+";
                    // Membuat perulangan untuk garis sebanyak kata terpanjang pada baris gender + 2 spasi
                    for (j = 0; j < gender + 2; j++)
                    {
                        cout << "-";
                    }
                }
                if (i == 8)
                {
                    cout << "+";
                    // Membuat perulangan untuk garis sebanyak kata terpanjang pada baris color + 2 spasi
                    for (j = 0; j < color + 2; j++)
                    {
                        cout << "-";
                    }
                }
                if (i == 9)
                {
                    cout << "+";
                    // Membuat perulangan untuk garis sebanyak kata terpanjang pada baris sleeve type + 2 spasi
                    for (j = 0; j < sleeve_type + 2; j++)
                    {
                        cout << "-";
                    }
                }
            }
            // penutup table (ujung kanan)
            cout << "+";
            cout << "" << endl;
        }

        // buat baris isi kolom atribut
        for (i = 0; i < 10; i++)
        {
            if (i == 0)
            {
                // Tampil nomor
                cout << "|" + isi[i] + "   ";
            }
            else if (i == 1)
            {
                // Tampil nama
                cout << "|" + isi[i];
                for (j = 0; j < idProduct - isi[i].length() + 2; j++)
                {
                    cout << " ";
                }
            }
            else if (i == 2)
            {
                // Tampil name
                cout << "|" + isi[i];
                for (j = 0; j < name - isi[i].length() + 2; j++)
                {
                    cout << " ";
                }
            }
            else if (i == 3)
            {
                // Tampil brand
                cout << "|" + isi[i];
                for (j = 0; j < brand - isi[i].length() + 2; j++)
                {
                    cout << " ";
                }
            }
            else if (i == 4)
            {
                // Tampil price
                cout << "|" + isi[i];
                for (j = 0; j < price - isi[i].length() + 2; j++)
                {
                    cout << " ";
                }
            }
            else if (i == 5)
            {
                // Tampil size
                cout << "|" + isi[i];
                for (j = 0; j < size - isi[i].length() + 2; j++)
                {
                    cout << " ";
                }
            }
            else if (i == 6)
            {
                // Tampil material
                cout << "|" + isi[i];
                for (j = 0; j < material - isi[i].length() + 2; j++)
                {
                    cout << " ";
                }
            }
            else if (i == 7)
            {
                // Tampil gender
                cout << "|" + isi[i];
                for (j = 0; j < gender - isi[i].length() + 2; j++)
                {
                    cout << " ";
                }
            }
            else if (i == 8)
            {
                // Tampil color
                cout << "|" + isi[i];
                for (j = 0; j < color - isi[i].length() + 2; j++)
                {
                    cout << " ";
                }
            }
            else if (i == 9)
            {
                // Tampil sleeve type
                cout << "|" + isi[i];
                for (j = 0; j < sleeve_type - isi[i].length() + 2; j++)
                {
                    cout << " ";
                }
            }
            // Membuat garis penutup table pada setiap atibut baris
            if (i == 9)
            {
                cout << "|";
                cout << "" << endl;
            }
        }

        // Membuat kerangka tabel pada baris terakhir
        if (idx == batas - 1)
        {
            cout << "+";
            for (i = 0; i < 10; i++)
            {
                // Jika i = 0
                if (i == 0)
                {
                    // Membuat perulangan sebanyak 4x("No" + 2 spasi)
                    for (j = 0; j < 4; j++)
                    {
                        cout << "-";
                    }
                }
                // Jika i = 1
                if (i == 1)
                {
                    cout << "+";
                    // Membuat perulangan sebanyak kata terpanjang pada baris nama + 2 spasi
                    for (j = 0; j < idProduct + 2; j++)
                    {
                        cout << "-";
                    }
                }
                // Jika i = 1
                if (i == 2)
                {
                    cout << "+";
                    // Membuat perulangan sebanyak kata terpanjang pada baris name + 2 spasi
                    for (j = 0; j < name + 2; j++)
                    {
                        cout << "-";
                    }
                }
                // Jika i = 3
                if (i == 3)
                {
                    cout << "+";
                    // Membuat perulangan sebanyak kata terpanjang pada baris brand + 2 spasi
                    for (j = 0; j < brand + 2; j++)
                    {
                        cout << "-";
                    }
                }
                // Jika i = 4
                if (i == 4)
                {
                    cout << "+";
                    // Membuat perulangan sebanyak kata terpanjang pada baris price + 2 spasi
                    for (j = 0; j < price + 2; j++)
                    {
                        cout << "-";
                    }
                }
                // Jika i = 5
                if (i == 5)
                {
                    cout << "+";
                    // Membuat perulangan sebanyak kata terpanjang pada baris size + 2 spasi
                    for (j = 0; j < size + 2; j++)
                    {
                        cout << "-";
                    }
                }
                // Jika i = 6
                if (i == 6)
                {
                    cout << "+";
                    // Membuat perulangan sebanyak kata terpanajgn pada baris material + 2 spasi
                    for (j = 0; j < material + 2; j++)
                    {
                        cout << "-";
                    }
                }
                // Jika i = 7
                if (i == 7)
                {
                    cout << "+";
                    // Membuat perulangan sebanyak kata terpanajgn pada gender + 2 spasi
                    for (j = 0; j < gender + 2; j++)
                    {
                        cout << "-";
                    }
                }
                // Jika i = 8
                if (i == 8)
                {
                    cout << "+";
                    // Membuat perulangan sebanyak kata terpanajgn pada color + 2 spasi
                    for (j = 0; j < color + 2; j++)
                    {
                        cout << "-";
                    }
                }
                // Jika i = 9
                if (i == 9)
                {
                    cout << "+";
                    // Membuat perulangan sebanyak kata terpanajgn pada sleeve type + 2 spasi
                    for (j = 0; j < sleeve_type + 2; j++)
                    {
                        cout << "-";
                    }
                }
            }
            // ujung kanan table
            cout << "+";
            cout << "" << endl;
        }
    }

    // Destruktor
    ~Table()
    {
    }
};